### PR TITLE
chore: fix the incorrect URL displayed during transaction confirmation

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

Up until now, the Connection request for deeplink based communication would present the sessionId instead of the URL. This would then be propagated to the `eth_sendTransaction` screen some some mixed outputs such as:
- sessionId for Android SDK
- sessionId for iOS SDK using deeplink procotol
- `https://` when a url was not provided and `window` was not accessible like NodeJS

This PR aims to streamline the usage of the metadata sent by dapps, SDK and not SDK related.

## **Related issues**
https://github.com/MetaMask/metamask-sdk/issues/862

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**

https://github.com/MetaMask/metamask-mobile/assets/61094771/cbead00e-f411-40d7-9c0b-0572361e6796

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by christopherferreira9** [comment]
@omridan159 is it possible to add a fallback for when the URL is not provided to the wallet such as a default message?

Screenshot 2024-06-11 at 15 53 45

cc: @andreahaku @abretonc7s

**Comment by christopherferreira9** [comment]
If we want to tackle the [corner case](https://github.com/MetaMask/metamask-mobile/pull/9914#issuecomment-2160973028) in a different issue since we enforce devs to provide a dapp url we're good to merge from the SDK. Checked:
- Web js dapp
- iOS SDK deeplink
- ReactNativeSDK (deeplink communication)
- NodeJS
- Android native SDK

**Comment by christopherferreira9** [comment]
The corder case stated in this [comment](https://github.com/MetaMask/metamask-mobile/pull/9914#issuecomment-2161026857) will be fixed in a different [PR](https://github.com/MetaMask/metamask-mobile/pull/9843) in order to get this one going since the case where a developer would be able to not provide a dappUrl is very minimal.

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 1]
```diff
@@ -127,8 +128,12 @@ const ApproveTransactionHeader = ({
       {origin && !isOriginDeepLink ? (
         <TagUrl
           testID={APPROVE_TRANSACTION_ORIGIN_PILL}
-          imageSource={faviconSource}
-          label={domainTitle}
+          imageSource={
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx:131]: Minor, but could we define the `imageSource` outside of the JSX?

```
const sdkIcon = sdkDappMetadata?.icon;
const sdkSource = sdkIcon && { uri: sdkIcon };
const imageSource = sdkSource ?? faviconSource;
```

File: app/components/Views/confirmations/components/TransactionReview/index.js [hunk 2]
```diff
@@ -528,6 +534,10 @@ class TransactionReview extends PureComponent {
               url={url}
               from={from}
               asset={transaction?.selectedAsset}
+              sdkDappMetadata={{
+                url: currentConnection?.originatorInfo?.url,
```
Comment (matthewwalsh0) [inline app/components/Views/confirmations/components/TransactionReview/index.js:538]: Minor, could we define the `sdkDappMetadata` outside of the JSX and without duplicating the full property chain?

```
const originatorInfo = currentConnection?.originatorInfo;
const sdkDappMetadata = { url: originatorInfo?.url, icon: originatorInfo?.icon };
```
```

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by OGPoyraz — DISMISSED** [review]

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 3]
```diff
@@ -122,13 +123,19 @@ const ApproveTransactionHeader = ({
...
+
   return (
     <View style={styles.transactionHeader}>
       {origin && !isOriginDeepLink ? (
         <TagUrl
           testID={APPROVE_TRANSACTION_ORIGIN_PILL}
-          imageSource={faviconSource}
-          label={domainTitle}
+          imageSource={imageSource}
+          label={sdkDappMetadata?.url || 'Unknown'}
```
Comment (andreahaku) [inline app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx:138]: this needs to be part of the strings file to be translated as it is shown to the user.

Comment (andreahaku) [reply app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx:138]: please check if it's the same for `app/components/Views/AccountConnect/AccountConnect.tsx`

File: app/components/Views/confirmations/components/TransactionReview/index.js [hunk 4]
```diff
@@ -524,8 +525,19 @@ class TransactionReview extends PureComponent {
...
+
+    const sdkConnections = SDKConnect.getInstance().getConnections();
+
+    const currentConnection = sdkConnections[origin ?? ''];
+
     const styles = this.getStyles();
 
+    const originatorInfo = currentConnection?.originatorInfo;
+    const sdkDappMetadata = {
+      url: originatorInfo?.url ?? 'Unknown',
```
Comment (andreahaku) [inline app/components/Views/confirmations/components/TransactionReview/index.js:537]: same here

**Review by andreahaku — COMMENTED** [review]
text that is part of the UI needs to be part of the translation files. Please fix it. Thanks

**Review by andreahaku — DISMISSED** [review]
LGTM

**Review by matthewwalsh0 — DISMISSED** [review]

**Comment by christopherferreira9** [comment]
Latest commits add a regression to WC transactions.

https://github.com/MetaMask/metamask-mobile/assets/104831203/33fb36e5-093b-4784-9215-c5133b00e32b

**Comment by christopherferreira9** [comment]
Hi @OGPoyraz @andreahaku @matthewwalsh0 @MetaMask/mobile-platform could you please assist with reviewing this PR? It has been reviewed and was approved before but after QA'ing it needed some extra checks on it.

**Review by andreahaku — APPROVED** [review]
LGTM

**Review by OGPoyraz — APPROVED** [review]

File: app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 5]
```diff
@@ -97,8 +99,10 @@ const ApproveTransactionHeader = ({
       title = getUrlObj(
         (origin as string).split(AppConstants.MM_SDK.SDK_REMOTE_ORIGIN)[1],
       ).origin;
+    } else if (url || currentEnsName) {
+      title = prefixUrlWithProtocol(currentEnsName || url || '');
```
Comment (kylanhurt) [inline app/components/Views/confirmations/components/ApproveTransactionHeader/ApproveTransactionHeader.tsx:103]: @omridan159 This line

**Pull request merged**